core/src/lib.rs

```diff
@@ -114,3 +173,47 @@ pub mod prelude {
         wsv::{World, WorldStateView},
     };
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
You mean create `role` module in `iroha_core`? Because if you are taking about `iroha_data_model` it's opposite of what i did before.

 https://github.com/hyperledger/iroha/pull/3917#discussion_r1335508283

Reply:
Yes, I mean `role` module in the same file, so it's `iroha_core`

Reply:
Do we nest `tests` modules? I haven't seen this before through codebase.

Reply:
Hm... Well I just think, that keeping tests closer to the code they test, is a good idea